src/App.tsx

```diff
@@ -14,11 +14,135 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[];
+  isVisible: boolean;
+  isReversed: boolean;
+  sortBy: 'name' | 'length' | null;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    isReversed: false,
+    sortBy: null,
+  };
+
+  start = () => {
+    this.setState({ isVisible: true });
+  };
+
+  reverse = () => {
+    this.setState((state) => (
+      { isReversed: !state.isReversed }
+    ));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({
+      sortBy: 'name',
```

Comment:
after introducing an `enum`
```suggestion
      sortBy: sortBy.name,
```